src/todo.ts

```diff
@@ -0,0 +1,123 @@
+﻿/**
+* Interface that carries attributes about items in the list including;
+* Date Created
+* Description of Task
+* Date to due by
+* Unique ID for the task
+* Whether the task is done or not
+*/
+export interface ListItem {
+    createdAt: Date;
+    description: string;
+    dueAt: Date;
+    id: number;
+    isDone: boolean;
+}
+
+// class that is use to create todo lists
+export class Todo {
+    public list: ListItem[];
+    public idNumber: number;
+    constructor() {
+        this.list = [];
+        this.idNumber = 0;
+    }
+
+    // Function to add items
+    public addItem(des: string, dateString: string, done: boolean = false) {
+        const today = new Date();
+        const dueDate = new Date(dateString);
+        if (dueDate instanceof Date && !isNaN(dueDate.valueOf())) {
+            const listItem = {
+                createdAt: today,
+                description: des,
+                dueAt: dueDate,
+                id: this.idNumber,
+                isDone: done,
+            };
+            this.idNumber++;
+            this.list.push(listItem);
+            return listItem;
+        } else {
+            throw new Error("The date entered is not valid.");
+        }
+    }
+
+    // Function to remove an Item given the ID
+    public removeItem(id: number): ListItem {
+        var item: ListItem;
+        const len: number = this.list.length;
+        for (let i = 0; i < len; i++) {
+            if (this.list[i].id == id) {
+                item = this.list[i];
+                this.list.splice(i,1);
+            }
+        }
+        return item;
+    }
+
+    // Prints out Info on each TODO item
+    public getListItems(): number[] {
+        const listIDs: number[] = [];
+        for (const item of this.list) {
+            const stat: string = (item ? "Completed" : " Not completed");
+            listIDs.push(item.id);
+            console.log(
+                "ID: " + item.id +
+                ", Description: " + item.description +
+                ", Created Date: " + (item.createdAt.getMonth() + 1) + "/" +
```

Comment:
Notice that you are repeating code here.  Maybe it would be easier to create a method that returns that string when given a date?  Always keep an eye out for things like this. Remember the DRY principle (Don't Repeat Yourself).